src/components/Calculator.js

```diff
@@ -1,34 +1,73 @@
 import React from 'react';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+    this.clickHandler = this.clickHandler.bind(this);
+    this.displayResult = this.displayResult.bind(this);
+  }
+
+  clickHandler(e) {
+    const { total, next } = this.state;
+    if (!total && !next && e.target.classList[0] === 'arithmetic_operand') {
+      return;
+    }
+    this.setState((objName) => calculate(objName, e.target.dataset.buttonName));
+  }
```

Comment:
**Optional Changes**
- [ ] Since `Calculator.js` is a child of `App.js`, the state and clickhandler function could be implemented in App.js and passed to Calculator.js as props. This will allow avoid writing redundant code for any App.js child should you want to access the same state or function.